gloo/allreduce.cc

```diff
@@ -216,13 +216,21 @@ void ring(
   //
   const size_t numSegments = roundUp(
       std::max(
-          (totalBytes + (segmentBytes - 1)) / segmentBytes,
+          (totalBytes + (tmpSegmentBytes - 1)) / tmpSegmentBytes,
           (size_t)context->size * 2),
       (size_t)context->size);
   GLOO_ENFORCE_EQ(numSegments % context->size, 0);
   GLOO_ENFORCE_GE(numSegments, context->size * 2);
   const size_t numSegmentsPerRank = numSegments / context->size;
```

Comment:
does it matter if `numSegments % context->size != 0`?

Reply:
The `roundUp` function returns the closest multiple of the second argument.

There is an additional `GLOO_ENFORCE` that checks that as an extra precaution / runtime check / piece of documentation.